tests/unit/test_charm.py

```diff
@@ -90,3 +91,63 @@ def test_ingress_media_removed(monkeypatch: MonkeyPatch) -> None:
     # when running update_config
     # when remove ingress-media relation
     assert mock_irc.reconcile.call_count == 2
+
+
+def test_ingress_media():
+    """
+    arrange: start the charm with all integrations and commands mocked.
+    act: add ingress-media relation.
+    assert: The media url points to the ingress relation data URL.
+    """
+    mock_irc = MagicMock()
+    with patch(
+        "charm.IRCBridgeService",
+        return_value=mock_irc,
+    ), patch("charm.MatrixObserver", return_value=MagicMock()):
+        ctx = testing.Context(IRCCharm)
+        config = {"bot_nickname": "testbot", "bridge_admins": "admin:example.com"}
+        database_data = {
+            "database": "ircbridge",
+            "endpoints": "postgresql-k8s-primary.local:5432",
+            "password": "123",
+            "username": "user1",
+        }
+        matrix_auth_data = {"shared_secret_id": "123", "homeserver": "https://example.com/"}
+        relations = [
+            testing.Relation(
+                id=1,
+                endpoint="database",
+                interface="postgresql_client",
+                remote_app_data=database_data,
+            ),
+            testing.Relation(
+                id=2,
+                endpoint="matrix-auth",
+                interface="matrix_auth",
+                remote_app_data=matrix_auth_data,
+            ),
+        ]
+        secrets = [testing.Secret(id="123", tracked_content={"shared-secret-content": "abc"})]
+        state = testing.State(leader=True, config=config, relations=relations, secrets=secrets)
+        out = ctx.run(ctx.on.config_changed(), state)
+        assert out.unit_status == testing.ActiveStatus()
+        assert mock_irc.reconcile.called
+        args, _ = mock_irc.reconcile.call_args
+        assert args[0].media_external_url == "http://192.0.2.0:11111"
+
+        media_url = "https://media/"
+        relations.append(
+            testing.Relation(
+                id=3,
+                endpoint="ingress-media",
+                interface="ingress",
+                remote_app_data={"ingress": json.dumps({"url": media_url})},
+            )
+        )
+        state = testing.State(leader=True, config=config, relations=relations, secrets=secrets)
+        out = ctx.run(ctx.on.config_changed(), state)
+
+        assert out.unit_status == testing.ActiveStatus()
+        assert mock_irc.reconcile.called
```

Comment:
Just a question: why dont we use mock_irc.assert_called_once() here?

Reply:
Reconcile is called twice :) But good catch, now I changed it to asser the call_count